File: +LoadData/+ismrmrd/+GE/+Functions/anonymise_mat.m
function anonymise_mat(filename_mat4anon)
%ANONYMISE_MAT Remove patient data from header structure in matlab mat file
% anonymise_mat(fname)
%  fname   Matlab mat file containing field h 
%
%  02/2021 Rolf Schulte
if (nargin<1), help(mfilename); return; end


%% input
if ~isempty(regexpi(filename_mat4anon,'\.mat$'))
    filename_mat4anon = filename_mat4anon(1:end-4); 
end
if ~exist([filename_mat4anon '.mat'],'file')
    error('fname (=''%s'') not found',[filename_mat4anon '.mat']);
end


%% load mat file
load([filename_mat4anon '.mat']);


%% anonymise h
if ~exist('h','var')
    error('header structure h not found'); 
end
h = header_anonymise(h);


%% save anoymised file
save([filename_mat4anon '_anon.mat']);


end      % anonymise_mat.m 

File: +LoadData/+GE_Recon/+fidall/bin/anonymise_mat.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
"""
Anonymise Matlab mat files
remove patient data from h (header) structure 
Created: 7/2025
@author: Rolf Schulte
"""

import sys
import sonofrecon as sor
import subprocess


# set parameters
fidalldir = sor.dir('fidall')
dofifo = sor.dofifo()        # use mascri running in background

# print help
if len(sys.argv)<2:
  print('anonymise_mat.py fname')
  print('fname of Matlab mat file containing h structure')
  sys.exit(-1)

# get input parameters
fname = str(sys.argv[1])

# actual execution of matlab-compiled program
cmd = ['anonymise_mat',fname]
if dofifo:
  sor.run_fifo(cmd)
else:
  sor.run(cmd)

sys.exit(0)

